src/mds/Server.cc

```diff
@@ -3534,10 +3544,57 @@ void Server::journal_allocated_inos(MDRequestRef& mdr, EMetaBlob *blob)
 		      mds->inotable->get_projected_version());
 }
 
-void Server::apply_allocated_inos(MDRequestRef& mdr, Session *session)
+void Server::try_delegate_inos(MDRequestRef& mdr, bool dirty) {
+  const cref_t<MClientRequest> &req = mdr->client_request;
+
+  if (req->get_op() == CEPH_MDS_OP_CREATE && mdr->tracei) {
+    if (mdr->did_early_reply) {
+      dout(20) << __func__ << ": inos probably delegated during early reply" << dendl;
+      return;
+    }
```

Comment:
Maybe we can just remove this check.

Just in case when doing `early reply` the session was dirty and then try to do it again in the safe reply. And also it harmless to delegate the inos twice.